File: scrapers/images/jaodb.pl
{
	searchURL => 'http://www.jaodb.com/db/sitemap.asp',
	searchStart => 0,
	searchEnd => 0,
	searchIncrement => 1,

	urlExtractor => scraper {
		process '//a', 'urls[]' => '@href';
	},

	cleanURL => sub {
		my $url = shift;
		return $url =~ /(\d+)/ ?
			"http://www.jaodb.com/db/ItemDetail.asp?item=$1" :
			"";
	},

	extractPageID => qr/(\d+)/,

	pageExtractor => scraper {
		process '//center/a[contains(@href,"LargeImage")]', source_image => '@href';
		process '//img[contains(@src,"F06")]', small_image => '@src';
		process '//a[contains(@href,"ItemDetail")]', 'urls[]' => '@href';
	},

	cleanPage => sub {
		my $item = shift;

		if ( !$item->{'source_image'} ) {
			$item->{'source_image'} = $item->{'small_image'};
		}

		delete $item->{'small_image'};

		$item->{'source_image'} =~ s!.*Images!http://www.jaodb.com/db/Images!;

		return $item;
	},

	extractImageFile => qr/([^\/]+.jpg)$/
}

File: scrapers/extractedimage/jaodb.js
exports.extract = {
	artist: ["//td[contains(font/text(), 'Artist:')]/following-sibling::td[1]",
		text => text.replace(/(?: \(.*?\))?\s+[—-]\s+(\S+)$/, " ($1)")],
	title: "//td[contains(font/text(), 'Title')]/following-sibling::td[1]",
	date: ["//td[contains(font/text(), 'Date')]/following-sibling::td[1]",
		text => text.replace(/^(\d+).*/, "$1")],
	description: "//td[contains(font/text(), 'Notes')]/following-sibling::td[text()]",
	source_image: ["//center/a[contains(@href,'LargeImage')]/@href || //center/img[contains(@src,'Images')][1]/@src",
		text => text.replace(/.*Images/, "http://www.jaodb.com/db/Images")]
};

exports.genURL = id => "http://www.jaodb.com/db/ItemDetail.asp?item=" + id;
